### PR TITLE
0 byte transfers no longer show as fail cases

### PR DESCRIPTION
**Pull request by bridgesr**
# PR checklist

Tick what applies before you raise. Delete anything that doesn't.

## Correctness
- [x] Does what the ticket asks for
- [x] Handles the edge cases (empty, null, zero, large inputs), not just the happy path
- [x] Errors are handled, not swallowed
- [x] New logic has tests, including the failure cases
- [x] Tests assert something real, not just run the code

## Keeping it simple
- [x] Doesn't rebuild something that already exists in the codebase
- [x] No more complex or slower than it needs to be
- [x] No leftover debug logging or commented-out code

## Easy to miss (a green pipeline won't flag these)
- [x] One logical change, not a pile of unrelated stuff

### If you touched the backend
- [x] Schema changes have a migration, and the Down() doesn't drop anything it shouldn't
- [x] New app settings are wired into the fa-config templates
- [x] Secrets use Key Vault references, not plain text
- [x] New outbound HTTP clients go through the shared resilience handler (AddResiliencePolicyHandler), rather than hand-rolling retries
- [x] New calls to an external service (DDEI, Egress, NetApp) have client tests against a WireMock stub (for the serialisation, auth and error handling) and are covered by the integration tests that run against the real dev services
- [x] Ran dotnet format, since CI builds and tests but doesn't check formatting

### If you touched the UI
- [x] Ran the e2e/ suite locally (the PR build only runs the mocked tests; the full suite runs post-merge and gates the deploy)
- [x] API changes are reflected in the MSW handlers in src/mocks and the matching zod schema, so the mocked tests aren't passing against a contract that's gone
- [x] New config or feature flags are wired into the VITE build variables, not just added in code
- [x] Data fetches handle the loading, empty and error states, not just success, and match how the rest of the app already does it
- [x] Ran prettier (npm run prettier:format), since CI lints but doesn't check formatting
- [x] Checked accessibility (keyboard nav, screen reader, sensible labels), since the Lighthouse run is manual and won't flag it for you

### CODE REVIEW

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Helpers/TransferResultProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            {
+                await context.Entities.CallEntityAsync(
+                    entityId,
+                    isRetry ? nameof(TransferEntityState.AddFailedRetryItem)
+                            : nameof(TransferEntityState.AddFailedItem),
+                    result.FailedItem);
+
+                telemetryEvent.TotalFilesFailed++;
+            }
+        }
```
Comment (brad-cps) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Helpers/TransferResultProcessor.cs:86]: The three branches don't cover the whole space and anything that misses all of them is dropped with no record at all. I'd consider adding an `else` that at minimum records the unclassifiable result.

File: backend/CPS.ComplexCases.FileTransfer.API/Durable/Helpers/TransferResultProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public static async Task ProcessAsync(
+        TaskOrchestrationContext context,
+        EntityInstanceId entityId,
+        TransferResult[] results,
+        TransferOrchestrationEvent telemetryEvent,
+        bool isRetry = false)
+    {
+        foreach (var result in results)
+        {
+            if (result != null && result.IsSkipped && result.SkippedItem != null)
```
Comment (brad-cps) [inline backend/CPS.ComplexCases.FileTransfer.API/Durable/Helpers/TransferResultProcessor.cs]: Minor point. The repeated `result != null` in every branch reads better as a `if (result is null) continue;` guard at the top of the loop... with a sequential `await` per result.

**Review by brad-cps — DISMISSED** [review]
Approved with minor comment

**Review by brad-cps — APPROVED** [review]
LGTM. Some Cognitive Complexity warning, but guess these can be resolved separately.
